src/commands/update.ts

```diff
@@ -1,73 +1,79 @@
-      return dependencies.indexOf(module.name) > -1;
-    } else {
-      return true;
-    }
-  });
+export class UpdateSubcommand extends Subcommand {
+  public signature = "update [deps_location] [module]";
+  public description =
+    "Update all dependencies in the `deps.ts` file in your CWD, or specify certain modules to update or a location to a dependency file.";
 
-  if (modules.length === 0) {
-    ConsoleLogger.error(
-      "Modules specified do not exist in your dependencies.",
+  public async handle() {
+    const depsLocation = Deno.args.find((arg) => arg.indexOf(".ts") > -1) ??
+      "deps.ts";
+    const dependencies = Deno.args.filter((arg) =>
+      arg.indexOf(".ts") === -1 && arg.indexOf("update") === -1
+    ); // Line doesnt allow us to get a n number of args whereby they can be optional
+    // Create objects for each dep, with its name and version
+    const allModules = await ModuleService.constructModulesDataFromDeps(
+      depsLocation,
     );
-    Deno.exit(1);
-  }
-
-  // Check for updates and rewrite `deps.ts` if needed
-  ConsoleLogger.info("Checking if your modules can be updated...");
-  const usersWorkingDir: string = Deno.realPathSync(".");
-  let depsWereUpdated = false;
-  let depsContent: string = new TextDecoder().decode(
-    Deno.readFileSync(usersWorkingDir + "/deps.ts"),
-  ); // no need for a try/catch. The user needs a deps.ts file
+    const modules = allModules.filter((module) => {
+      if (dependencies.length) { // only return selected modules of selecting is set
+        return dependencies.indexOf(module.name) > -1;
+      } else {
+        return true;
+      }
+    });
 
-  // Update the file content
-  modules.forEach((module) => {
-    // only re-write modules that need to be updated
-    if (module.importedVersion === module.latestRelease) {
-      return;
-    }
-    if (module.std) {
-      depsContent = depsContent.replace(
-        "std@" + module.importedVersion + "/" + module.name,
-        "std@" + module.latestRelease + "/" + module.name,
-      );
-    } else {
-      let newImportUrl = module.importUrl;
-      newImportUrl = newImportUrl.replace(
-        module.importedVersion,
-        module.latestRelease,
+    if (modules.length === 0) {
+      ConsoleLogger.error(
+        "Modules specified do not exist in your dependencies.",
       );
-      depsContent = depsContent.replace(module.importUrl, newImportUrl);
+      Deno.exit(1);
     }
-    ConsoleLogger.info(
-      module.name + " was updated from " + module.importedVersion + " to " +
-        module.latestRelease,
-    );
-    depsWereUpdated = true;
-  });
 
-  // Re-write the file
-  Deno.writeFileSync(
-    usersWorkingDir + "/deps.ts",
-    new TextEncoder().encode(depsContent),
-  );
+    // Check for updates and rewrite `deps.ts` if needed
+    ConsoleLogger.info("Checking if your modules can be updated...");
+    const usersWorkingDir: string = Deno.realPathSync(".");
```

Comment:
Since we are inside an `async function`, let's use the promised based API where possible.
I would argue if here a `Deno.cwd()` wouldn't be better, but to not affect the current implementation, `await Deno.realPath(".");`

Reply:
Sync code (even if inside an async method) is faster than async, so i'd be quite reluctent to switch this, with the goal of speed being in mind

(A **basic** example I have showed using an async API method was nearly 3 slower, albeit this could differ but i guess the fact its an decrease in speed shows)

Reply:
Wait, what? Is it still that?

Reply:
oh wtf.

Reply:
> Wait, what? Is it still that?

not entirely sure what you're saying here?

But i did have a typo in my original comment - i meant that using the async variant of a Deno method was nearly 3 times slower than the sync variant:

```
sync: 0.33ms avg
async: 1.2ms avg
```

Granted it is small, but dmm does a lot of fs access stuff (and fetch requests), so the overall speed could be impacted in a somewhat noticable way

Reply:
i'm curious to know what two async calls back to back and two sync calls back to back would result in since one is blocking and the other isn't. imma test it

Reply:
I understood! I'm saying "what" in terms of it's stupid this is the case. And I was wondering if it still applies in the current Deno version.

Reply:
> I understood! I'm saying "what" in terms of it's stupid this is the case. And I was wondering if it still applies in the current Deno version.

gotcha, yeah this was tested on 1.10.2 (though 1.10.3 was just released an hour ago)